### PR TITLE
adds enum extends to report element part of SCI-1003

### PR DESCRIPTION
**Pull request by ZmagoD**
part of http://gitlab.biosistemika.com/scinote/electronic_signatures/merge_requests/4

https://biosistemika.atlassian.net/browse/SCI-1070

also fixes https://biosistemika.atlassian.net/browse/SCI-1100

### CODE REVIEW

**Comment by Ducz0r** [comment]
Since I'll not be present next week, I'm unassigning myself off this PR.

File: app/helpers/reports_helper.rb [hunk 1]
```diff
@@ -1,99 +1,95 @@
...
-  elsif element.result_comments?
-    locals[:result] = element.element_reference
-    locals[:order] = element.sort_order
-  elsif element.project_activity?
-    # TODO
-  elsif element.project_samples?
-    # TODO
-  end
+    ReportElement.type_ofs.keys.each do |type|
+      next unless element.send("#{type}?")
```
Comment (okriuchykhin) [inline app/helpers/reports_helper.rb]: Can you change it to 'public_send'?

File: app/models/report_element.rb [hunk 2]
```diff
@@ -59,44 +46,17 @@ def comments?
...
-    elsif step_checklist?
-      self.checklist_id = ref_id
-    elsif step_asset?
-      self.asset_id = ref_id
-    elsif step_table?
-      self.table_id = ref_id
+    ReportExtends::SET_ELEMENT_REFERENCES_LIST.each do |el_ref|
+      check = el_ref.check(self)
+      next unless check
+      self.send("#{el_ref.element}=", ref_id)
```
Comment (okriuchykhin) [inline app/models/report_element.rb]: Can you change it just to public_send("#{el_ref.element}=", ref_id) ?

File: app/helpers/reports_helper.rb (hunk 1 above)
Comment (okriuchykhin) [inline app/helpers/reports_helper.rb]: same here

**Pull request merged**